Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/CommonComponents/LocationDrawingsButton.tsx

```diff
@@ -0,0 +1,148 @@
+            && locationsWithErrors.get(loc?.ID).length > errors.length) {// fewer errors than before
+            handleRemoveLocationError(loc.ID);
+        } else if (errors.length > 0) {
+            const locationErrorsMap = new Map<number, string[]>();
+            locationErrorsMap.set(loc?.ID, errors);
+            handleAddLocationError(locationErrorsMap);
+        }
+    }
+
+    const handleShowDrawingsModal = (loc) => {
+        if (loc == undefined) return;
+        setSelectedLocation(loc);
+        setShowDrawingsModal(true);
+    };
+
+    return (
+        <div>
+            <LoadingScreen Show={pageState == 'loading'} />
+            <ServerErrorIcon Show={pageState == 'error'} Size={40} Label={'A Server Error Occurred. Please Reload the Application.'} />
+            {!multipleLocations
+            ? <>
+            <button
+                className={locationsWithErrors.has(props.Locations[0]?.ID) ? "btn btn-primary disabled" : "btn btn-primary"}
+                onClick={() => locationsWithErrors.has(props.Locations[0]?.ID) ? null : handleShowDrawingsModal(props.Locations[0])}
+                data-tooltip={"DrawingsModal"}
+                onMouseEnter={() => setHover('drawings')}
+                onMouseLeave={() => setHover('none')}
+                >Open {props.Locations[0]?.Name} Drawings
+            </button>
+            <ToolTip
+                Show={locationsWithErrors.has(props.Locations[0]?.ID) && hover === 'drawings'}
+                Theme={'dark'}
+                Position={'top'}
+                Target={"DrawingsModal"}
+                Zindex={9999}
+                > {locationsWithErrors.get(props.Locations[0]?.ID)?.map((e, i) => <p key={i}>{CrossMark} {e}</p>)}
+            </ToolTip>
+            </>
+            : <BtnDropdown
+                Label={'Open ' + props.Locations[0]?.Name + ' Drawings'}
+                Callback={() => handleShowDrawingsModal(props.Locations[0])}
+                TooltipContent={
+                    <>{locationsWithErrors.get(props.Locations[0]?.ID)?.map((e, i) => <p key={i}>{CrossMark} {e}</p>)}</>
+                }
+                ShowToolTip={locationsWithErrors.has(props.Locations[0]?.ID)}
+                Disabled={locationsWithErrors.has(props.Locations[0]?.ID)}
+                BtnClass={'btn-primary'}
+                Options={props.Locations.slice(1).map((loc, i) => ({
+                    Label: 'Open ' + loc?.Name + ' Drawings',
+                    Callback: () => handleShowDrawingsModal(loc),
+                    Disabled: locationsWithErrors.has(loc?.ID),
+                    ToolTipContent: <>{
+                        locationsWithErrors.get(loc?.ID)?.map((e, i) => <p key={i}>{CrossMark} {e}</p>)
+                    }</>,
+                    ShowToolTip: locationsWithErrors.has(loc?.ID),
+                    ToolTipLocation: "left",
+                    Key: i
+                }))}
```

Comment:
could we save the `options` as states that get returned from the ` LocationDrawingController.PagedSearch`? that would simplify error handling because rather than having to store the errors in a map to map back to the location you can just save `Label`, `Callback`, Disabled`, `ToolTipContent` and `ShowToolTip` in a single list and hand that into Options here